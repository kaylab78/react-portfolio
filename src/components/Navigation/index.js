import React from 'react';

function Navigation() {
    return (
        <header>
            <h2>
                <a data-testid="link" href="/">
                    Kayla Backus
                </a>
            </h2>
            <nav>
                <ul>
                    <li>
                        <a data-testid="about" href="#about">
                            About Me
                        </a>
                    </li>
                    <li>
                        <a data-testid="portfolio" href="#portfolio">
                            Portfolio
                        </a>
                    </li>
                    <li>
                        <a data-testid="contact" href="#contact">
                            Contact Me
                        </a>
                    </li>
                    <li>
                        <a data-testid="resume" href="#resume">
                            Resume
                        </a>
                    </li>
                </ul>
            </nav>
        </header>
    );
}

export default Navigation;