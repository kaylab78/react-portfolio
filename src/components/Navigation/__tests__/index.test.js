import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Navigation from '..';

afterEach(cleanup);

describe('Navigation component', () => {
    it('renders', () => {
        render(<Navigation />);
    });

    it('matches snapshot', () => {
        const { asFragment } = render(<Navigation />);
        expect(asFragment()).toMatchSnapshot();
    });
});

describe('links are visible', () => {
    it('inserts text into the links', () => {
        const { getByTestId } = render(<Navigation />);

        expect(getByTestId('link')).toHaveTextContent('Kayla Backus');
        expect(getByTestId('about')).toHaveTextContent('About Me');
        expect(getByTestId('portfolio')).toHaveTextContent('Portfolio');
        expect(getByTestId('contact')).toHaveTextContent('Contact Me');
        expect(getByTestId('resume')).toHaveTextContent('Resume');
    });
});