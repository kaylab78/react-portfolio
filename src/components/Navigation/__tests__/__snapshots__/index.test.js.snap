// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component matches snapshot 1`] = `
<DocumentFragment>
  <header>
    <h2>
      <a
        href="/"
      >
        Kayla Backus
      </a>
    </h2>
    <nav>
      <ul>
        <li>
          <a
            href="#about"
          >
            About Me
          </a>
        </li>
        <li>
          <a
            href="#portfolio"
          >
            Portfolio
          </a>
        </li>
        <li>
          <a
            href="#contact"
          >
            Contact Me
          </a>
        </li>
        <li>
          <a
            href="#resume"
          >
            Resume
          </a>
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
